tasks/collect-pyxis-params/collect-pyxis-params.yaml

```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-pyxis-params
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that collects pyxis configuration options from the data file
+  params:
+    - name: dataPath
+      type: string
+      description: Path to the JSON string of the merged data containing the pyxis configuration options to use
+      default: data.json
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  results:
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'. Default is 'production'
+    - name: secret
+      type: string
+      description: |
+        The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert
+  steps:
+    - name: collect-pyxis-params
+      image:
+        quay.io/hacbs-release/release-utils:48a1cd6c939eda25382472135be4afe2c9b4b703
+      script: |
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.pyxis | has("server")' $DATA_FILE) == false ] ; then
+            echo "production" > $(results.server.path)
+        else
+            jq -r '.pyxis.server' $DATA_FILE > $(results.server.path)
+        fi
+        if [ $(jq '.pyxis | has("secret")' $DATA_FILE) == false ] ; then
+            echo "No secret provided in pyxis key. Failing..."
```

Comment:
What if there isn't even a `pyxis` key in the json? It will fail the same, but then this text will not be accurate. So I wonder if maybe we should add a check for that earlier? Or else you could at least modify this text to say: "pyxis key or secret missing in data JSON file."

Reply:
I updated the echo